openfl/interface/collaborator.py

```diff
@@ -258,7 +260,7 @@ def calchash(data_path):
         sys.exit(1)
     with open(datasources_json_path, "r", encoding="utf-8") as file:
         data = file.read()
-    vds = DataSourcesJsonParser.parse(data)
+    vds = DataSourcesJsonParser.parse(data, check_dir_traversal=True)
```

Comment:
Instead, can we `check_dir_traversal` before calling this function? I don't think the class needs to know about whether it is in a valid subdirectory...